Tools/autotest/common.py

```diff
@@ -1305,113 +1353,114 @@ def test_arm_feature(self):
+            self.disarm_vehicle()
+            self.mav.wait_heartbeat()
+            self.start_test("Test disarming failure with ARMING_RUDDER=1")
+            self.set_parameter("ARMING_RUDDER", 1)
+            self.arm_vehicle()
+            if self.disarm_motors_with_rc_input():
                 self.progress("Failed to NOT ARM")
                 raise NotAchievedException()
             self.disarm_vehicle()
             self.mav.wait_heartbeat()
-            self.set_rc(arming_switch, 1000)
-            self.set_rc(interlock_channel, 1000)
-            if self.mav.mav_type is mavutil.mavlink.MAV_TYPE_HELICOPTER:
-                self.start_test("Test motor interlock enable can't be set while disarmed")
+            self.set_parameter("ARMING_RUDDER", 2)
+            if self.mav.mav_type in [mavutil.mavlink.MAV_TYPE_QUADROTOR,
+                                     mavutil.mavlink.MAV_TYPE_HELICOPTER,
+                                     mavutil.mavlink.MAV_TYPE_HEXAROTOR,
+                                     mavutil.mavlink.MAV_TYPE_OCTOROTOR,
+                                     mavutil.mavlink.MAV_TYPE_COAXIAL,
+                                     mavutil.mavlink.MAV_TYPE_TRICOPTER]:
+                self.start_test("Test arming failure with interlock enabled")
                 self.set_rc(interlock_channel, 2000)
-                channel_field = "servo%u_raw" % interlock_channel
-                interlock_value = self.get_parameter("SERVO%u_MIN" % interlock_channel)
-                tstart = self.get_sim_time()
-                while True:
-                    remaining = 20 - (self.get_sim_time_cached() - tstart)
-                    if remaining <= 0:
-                        break
-                    m = self.mav.recv_match(type='SERVO_OUTPUT_RAW',
-                                            blocking=True,
-                                            timeout=remaining)
-                    m_value = getattr(m, channel_field, None)
-                    if m_value is None:
-                        raise ValueError()
-                    self.progress("SERVO_OUTPUT_RAW.%s=%u want=%u" %
-                                  (channel_field, m_value, interlock_value))
-                    if m_value != interlock_value:
-                        raise NotAchievedException("Motor interlock was changed while disarmed")
-            self.set_rc(8, 1000)
+                if self.arm_motors_with_rc_input():
+                    self.progress("Failed to NOT ARM")
+                    raise NotAchievedException()
+                if self.arm_motors_with_switch(arming_switch):
+                    self.progress("Failed to NOT ARM")
+                    raise NotAchievedException()
+                self.disarm_vehicle()
+                self.mav.wait_heartbeat()
+                self.set_rc(arming_switch, 1000)
+                self.set_rc(interlock_channel, 1000)
+                if self.mav.mav_type is mavutil.mavlink.MAV_TYPE_HELICOPTER:
```

Comment:
I think I'd prefer factoring the test and overriding the factored component in the subclass.

Reply:
+1